cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,333 @@ func TestConstructor(t *testing.T) {
+	logger, logBuf := testutils.NewLogger()
+	factoryParams := ProcessorFactoryParams{
+		Topic:          topic,
+		Parallelism:    1,
+		SaramaConsumer: saramaConsumer,
+		BaseProcessor:  spanProcessor,
+		Logger:         logger,
+		Factory:        metrics.NullFactory,
+	}
 
-	undertest.partitionIDToState = map[int32]*consumerState{
-		partition: {
-			partitionConsumer: &partitionConsumerWrapper{
-				topic:             topic,
-				partition:         partition,
-				PartitionConsumer: &kmocks.PartitionConsumer{},
-			},
-		},
+	processorFactory, err := NewProcessorFactory(factoryParams)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	consumerParams := Params{
+		InternalConsumer:      saramaConsumer,
+		ProcessorFactory:      *processorFactory,
+		MetricsFactory:        metrics.NullFactory,
+		Logger:                logger,
+		DeadlockCheckInterval: 0,
 	}
 
-	undertest.Start()
+	consumer, err := New(consumerParams)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
 
-	mc.YieldMessage(msg)
-	isProcessed.Wait()
+	spanWriter.consumer = consumer
+	spanWriter.t = t
 
-	localFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.partitions-held",
-		Value: 1,
-	})
+	consumer.Start()
+	consumer.Ready()
 
-	mp.AssertExpectations(t)
-	// Ensure that the partition consumer was updated in the map
-	assert.Equal(t, saramaPartitionConsumer.HighWaterMarkOffset(),
-		undertest.partitionIDToState[partition].partitionConsumer.HighWaterMarkOffset())
-	undertest.Close()
+	t.Log("consumer is ready and wait message")
 
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.partitions-held",
-		Value: 0,
-	})
+	spanWriter.done.Wait()
 
-	partitionTag := map[string]string{"partition": fmt.Sprint(partition)}
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.messages",
-		Tags:  partitionTag,
-		Value: 1,
-	})
-	localFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.current-offset",
-		Tags:  partitionTag,
-		Value: int(msgOffset),
-	})
-	localFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
-		Name: "sarama-consumer.offset-lag",
-		Tags: partitionTag,
-		// Prior to sarama v1.31.0 this would be 0, it's unclear why this changed.
-		// v=1 seems to be correct because high watermark in mock is incremented upon
-		// consuming the message, and func HighWaterMarkOffset() returns internal value
-		// (already incremented) + 1, so the difference is always 2, and we then
-		// subtract 1 from it.
-		Value: 1,
-	})
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.partition-start",
-		Tags:  partitionTag,
-		Value: 1,
-	})
+	t.Logf("consumer all logs: %s", logBuf.String())
 }
 
-func TestSaramaConsumerWrapper_start_Errors(t *testing.T) {
-	localFactory := metricstest.NewFactory(0)
+func TestConsumerGroupForReturnError(t *testing.T) {
```

Comment:
The test name seems to suggest the consumer should return errors, although I couldn't find where that assertion is in this test?